components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1899,40 +1899,45 @@ mod confchange_cmd_metric {
     }
 }
 
-// Validate the request and the split keys
-pub fn extract_split_keys(
-    split_reqs: &BatchSplitRequest,
-    region_to_split: &Region,
-) -> Result<VecDeque<Vec<u8>>> {
-    if split_reqs.get_requests().is_empty() {
+pub fn validate_batch_split(req: &AdminRequest, region: &Region) -> Result<()> {
+    if !req.has_splits() || req.get_splits().get_requests().is_empty() {
```

Comment:
Redundant check for `has_splits`.

Reply:
Good catch.  get_splits() will return a default if not set.